src/wp-includes/meta.php

```diff
@@ -1798,3 +1830,69 @@ function get_object_subtype( $object_type, $object_id ) {
 	 */
 	return apply_filters( "get_object_subtype_{$object_type}", $object_subtype, $object_id );
 }
+
+/**
+ * Check if the object support database type.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $object_type Type of object.
+ *
+ * @return bool True mean the object support database type, false otherwise.
+ */
+function wp_check_object_support_meta_value_type( $object_type ) {
+	$object_type_supported = in_array( $object_type, array( 'post', 'comment', 'term', 'user' ), true );
+
+	/**
+	 * Filters if the object type support database type.
+	 *
+	 * Any object type declaring support database type must have the corresponding column in its meta table.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param bool $object_type_supported Support flag for the object type.
+	 * @param string $object_type Type of object.
+	 */
+	return apply_filters( '', $object_type_supported, $object_type );
+}
```

Comment:
Not sure why this is needed.

Reply:
This is to ensure that calling [`add_metadata` or `update_metadata`](https://developer.wordpress.org/reference/functions/update_metadata/) with a custom `$meta_type` wouldn't cause an SQL error if the schema of the custom metadata table hadn't been updated yet with the new column for the type.